src/circuit/gadgets/constraints.rs

```diff
@@ -242,6 +242,36 @@ where
     Ok(c)
 }
 
+/// Convert from Boolean to AllocatedNum
+pub fn convert_boolean_to_num<F: PrimeField, CS: ConstraintSystem<F>>(
+    mut cs: CS,
+    bit: &Boolean,
+) -> Result<AllocatedNum<F>, SynthesisError>
+where
+    CS: ConstraintSystem<F>,
+{
+    let num = AllocatedNum::alloc(cs.namespace(|| "Allocate num"), || match bit.get_value() {
+        Some(v) => {
+            if v {
+                Ok(F::one())
+            } else {
+                Ok(F::zero())
+            }
+        }
+        None => Ok(F::zero()),
```

Comment:
I don't think this right. You shouldn't pluck a zero out of thin air, if the boolean has no value. I think that should be a `SynthesisError::AssignmentMissing`.

Reply:
Done